lib/js/components/Headers/SectionHeader/SectionHeader.vue

```diff
@@ -0,0 +1,343 @@
+			margin-left: $space-xxs;
+		}
+	}
+
+	&.-size-s {
+		#{$root}__titleWrapper {
+			@include heading-s-default-bold;
+		}
+
+		#{$root}__iconLeft {
+			margin-right: $space-xxxs;
+		}
+
+		#{$root}__iconRight,
+		#{$root}__chevron {
+			margin-left: $space-xxxs;
+		}
+	}
+
+	&.-size-xs {
+		#{$root}__title {
+			padding: $space-xxxs 0;
+		}
+
+		#{$root}__titleWrapper {
+			@include heading-xs-default-bold-uppercase;
+		}
+
+		#{$root}__iconLeft {
+			margin-right: $space-xxxs;
+		}
+
+		#{$root}__iconRight,
+		#{$root}__chevron {
+			margin-left: $space-xxxs;
+		}
+	}
+
+	&__slotHorizontal {
+		display: none;
+
+		@media #{breakpoint-s()} {
+			display: block;
+		}
+	}
+
+	&__slotVertical {
+		display: block;
+		padding: 0 0 $space-xxxs;
+
+		@media #{breakpoint-s()} {
+			display: none;
+		}
+	}
+
+	&.-horizontal {
+		#{$root}__slotHorizontal {
+			display: block;
+		}
+
+		#{$root}__slotVertical {
+			display: none;
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import { SECTION_HEADER_MOBILE_LAYOUTS, SECTION_HEADER_SIZES } from './SectionHeader.consts';
+import DsIcon, { ICON_SIZES, IconItem, ICONS } from '../../Icons/Icon';
+import DsIconButton, { ICON_BUTTON_COLORS, ICON_BUTTON_SIZES } from '../../Buttons/IconButton';
+import DsDivider from '../../Divider';
+import { toRaw } from 'vue';
+
+export default {
+	name: 'SectionHeader',
+	components: {
+		DsIcon,
+		DsIconButton,
+		DsDivider,
+	},
+	props: {
+		expandable: {
+			type: Boolean,
+			default: false,
+		},
+		hideSlotWhenCollapsed: {
+			type: Boolean,
+			default: false,
+		},
+		iconLeft: {
+			type: Object as () => IconItem,
+			default: null,
+			validator(iconLeft: IconItem) {
+				return Object.values(ICONS).includes(toRaw(iconLeft));
+			},
+		},
+		iconRight: {
+			type: Object as () => IconItem,
+			default: null,
+			validator(iconRight: IconItem) {
+				return Object.values(ICONS).includes(toRaw(iconRight));
+			},
+		},
+		isExpanded: {
+			type: Boolean,
+			default: false,
+		},
+		info: {
+			type: Boolean,
+			default: false,
+		},
+		size: {
+			type: String,
+			default: SECTION_HEADER_SIZES.M,
+			validator(size) {
+				return Object.values(SECTION_HEADER_SIZES).includes(size);
+			},
+		},
+		title: {
+			type: String,
+			required: true,
+		},
+		supportingText: {
+			type: String,
+			default: null,
+		},
+		divider: {
+			type: Boolean,
+			default: true,
+		},
+		mobileLayout: {
+			type: String,
+			default: SECTION_HEADER_MOBILE_LAYOUTS.VERTICAL,
+			validator: (value) => Object.values(SECTION_HEADER_MOBILE_LAYOUTS).includes(value),
+		},
+	},
+	data() {
+		return {
+			ICONS: Object.freeze(ICONS),
+			ICON_SIZES: Object.freeze(ICON_SIZES),
+			ICON_BUTTON_SIZES: Object.freeze(ICON_BUTTON_SIZES),
+			ICON_BUTTON_COLORS: Object.freeze(ICON_BUTTON_COLORS),
+			SECTION_HEADER_MOBILE_LAYOUTS: Object.freeze(SECTION_HEADER_MOBILE_LAYOUTS),
+		};
+	},
+	computed: {
+		chevronRotation(): number | null {
+			return this.isExpanded ? 180 : null;
```

Comment:
```suggestion
			return this.isExpanded ? 180 : 0;
```

Reply:
if we use `null`, vue don't render this prop and use default value if we pass 0 and in future we change default value it blowup :D

Reply:
IDE is underlining the line where this property is used. Another thing is why we may want to change meaning of 0 for this prop, it would be counterintuitive.

Reply:
if we pass `0` we got an error from the validator because we have `return [90, 180, 270].includes(value)`

Reply:
no longer underlined, changed to `undefined`